src/fe-common/core/fe-windows.c

```diff
@@ -400,71 +440,112 @@ WINDOW_REC *window_find_item(SERVER_REC *server, const char *name)
 
 int window_refnum_prev(int refnum, int wrap)
 {
-	GSList *tmp;
-	int prev, max;
+	GSequenceIter *iter, *end;
 
-	max = prev = -1;
-	for (tmp = windows; tmp != NULL; tmp = tmp->next) {
-		WINDOW_REC *rec = tmp->data;
+	iter = g_sequence_search(windows_seq, GINT_TO_POINTER(refnum-1), (GCompareDataFunc)window_refnum_lookup, NULL);
```

Comment:
Bar the different traversal direction the `_prev` and `_next` variants look pretty similar, can you perhaps merge the two under a `window_refnum_direction` and then make `window_refnum_prev` and `window_refnum_next` dumb wrappers over that function ?